var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -576,11 +644,25 @@ def edit(self, spec, prefix):
                     libs += ["-lcufft", "-lcublas"]
 
             cuda_arch = spec.variants["cuda_arch"].value[0]
-            if cuda_arch:
-                gpuver = {"35": "K40", "37": "K80", "60": "P100", "70": "V100"}[cuda_arch]
+            gpuver = gpu_map[cuda_arch]
+            if cuda_arch == "35" and spec.satisfies("+cuda_arch_35_k20x"):
+                gpuver = "K20X"
 
-                if cuda_arch == "35" and spec.satisfies("+cuda_arch_35_k20x"):
-                    gpuver = "K20X"
+        if spec.satisfies(+rocm) and spec.satisfies("@2022:"):
+            libs += [
+                "-L{}".format(spec["rocm"].libs.directories[0]),
+                "-L{}/stubs".format(spec["rocm"].libs.directories[0]),
+                "-lhipblas",
+                "-lhipfft",
+                "-lstdc++",
```

Comment:
I don't know the reason why you need to explicitly instruct the room dependency like this but FIY

```bash
>>> from spack.spec import Spec
>>> s = Spec("hwloc") # just a random package with a dependency
>>> s.concretize()
>>> dep = s["ncurses"]
>>> dep.libs.search_flags
'-L/Users/ialberto/spack/opt/spack/darwin-ventura-m2/clang-16.0.3/ncurses-6.4-pylthzrstz72hhppg4oiyr3glvziyyur/lib'
>>> dep.libs.link_flags
'-lncurses -ltinfo -lncursesw -ltinfow'
>>> dep.libs.ld_flags
'-L/Users/ialberto/spack/opt/spack/darwin-ventura-m2/clang-16.0.3/ncurses-6.4-pylthzrstz72hhppg4oiyr3glvziyyur/lib -lncurses -ltinfo -lncursesw -ltinfow'
```